### PR TITLE
Improve cheatsheet update machinery

### PR DESCRIPTION
**Pull request by pokey**
- Sorts `defaults.json` for diff stability
- Adds Talon-side action for updating `defaults.json`

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)

### CODE REVIEW

File: cursorless-nx/libs/cheatsheet/src/lib/data/sampleSpokenFormInfos/defaults.json [hunk 1]
```diff
@@ -5,252 +5,266 @@
 			"id": "actions",
 			"items": [
 				{
-					"id": "scrollToBottom",
```
Comment (pokey) [inline cursorless-nx/libs/cheatsheet/src/lib/data/sampleSpokenFormInfos/defaults.json:8]: alphabetised / added things which had been forgotten from previous updates.  Remember this file is just for convenience during local cheatsheet development

File: src/core/Cheatsheet.ts [hunk 2]
```diff
@@ -111,10 +111,38 @@ export default class Cheatsheet {
       "defaults.json"
     );
 
-    await writeFile(defaultsPath, JSON.stringify(spokenFormInfo, null, "\t"));
+    const outputObject = produce(spokenFormInfo, (draft) => {
+      draft.sections = sortBy(draft.sections, "id");
+      draft.sections.forEach((section) => {
+        section.items = sortBy(section.items, "id");
+      });
+    });
```
Comment (pokey) [inline src/core/Cheatsheet.ts:119]: I think [immer](https://immerjs.github.io/immer/) is a nice way to handle immutable data structures.  Planning to start using it if no objections

thanks for the pointer @SimeonC

**Review by AndreasArvidsson — APPROVED** [review]

File: src/core/Cheatsheet.ts (hunk 2 above)
Comment (auscompgeek) [reply src/core/Cheatsheet.ts:119]: 👍 It does seem like it's more suited to our use case than immutable.js which seems to be the other choice here. I found an article comparing the two: https://blog.logrocket.com/immer-and-immutable-js-how-do-they-compare/

Comment (pokey) [reply src/core/Cheatsheet.ts:119]: That's what they do [in the docs](https://immerjs.github.io/immer/#with-immer) 🤷‍♂️

**Pull request merged**